Add 0.10.3 of secretgen-controller

neil-hickey opened the pull request:
- This simplfies the process of ingesting secretgen as a package
- TCE just needs the published artifacts from SGC github now

Signed-off-by: Neil Hickey <nhickey@vmware.com>

## What this PR does / why we need it
This PR simplifies the process for secretgen controller PackageCR management from 0.10.3 onwards.

It is also a signal that the PackageCR management will be done by the carvel team in the OSS repo 

https://github.com/vmware-tanzu/carvel-secretgen-controller/tree/develop/config

## Describe testing done for PR
The release process for SGC runs our e2e tests against the PackageCR installed via a PackageInstallCR

## Special notes for your reviewer
The values schema is different now - this could have implications on any downstream consumers of TCE

For the values schema... checkout https://github.com/vmware-tanzu/carvel-secretgen-controller/blob/develop/config/package-bundle/config/values-schema.yml

neil-hickey committed:
Add 0.10.3 of secretgen-controller

- This simplfies the process of ingesting secretgen as a package
- TCE just needs the published artifacts from SGC github now

Signed-off-by: Neil Hickey <nhickey@vmware.com>

cppforlife commented on addons/packages/secretgen-controller/vendir.yml:
```diff
@@ -0,0 +1,13 @@
+          slug: vmware-tanzu/carvel-secretgen-controller
+          tag: v0.10.3
+          assetNames: ["package.yml"]
+          checksums:
+            package.yml: 0fdcfbaff66b5bbfea0cef9b261c60552417ee008c5cb852e7604e5ea9215605
```
btw since we have checksums within release notes they should be automatically picked up.

neil-hickey replied:
should... but it doesn't seem to work for me.

```
vendir sync
Fetching: 0.10.3 + . (github release vmware-tanzu/carvel-secretgen-controller@v0.10.3)

vendir: Error: Syncing directory '0.10.3':
  Syncing directory '.' with github release contents:
    Finding checksums in release notes:
      Expected to find sha256 checksum for file 'package.yml'
```

neil-hickey committed:
Remove manual checksum

The release notes of SGC have been fixed so vendir can automatically parse them

seemiller reviewed (commented):
@neil-hickey Would it be possible to include the [carvel-secretgen-controller](https://github.com/vmware-tanzu/carvel-secretgen-controller)/[config](https://github.com/vmware-tanzu/carvel-secretgen-controller/tree/develop/config)/[package](https://github.com/vmware-tanzu/carvel-secretgen-controller/tree/develop/config/package)/README.md  file ?

neil-hickey commented:
👋 Hey @seemiller , what exactly is the README used for? We were chatting about it, and it doesn't feel right to have the README as an artifact that we produce.

All the information in a README should be derivable from the PackageCR itself, perhaps some tool could exist / be created that generates the README from the PackageCR. 

Some the questions boil down to:

1. Do we really need a README.md?
2. Can the readme just live at the top level (not associated with a particular version)?

seemiller commented:
The readme is basically a quick reference for the package within TCE. If you're browsing the codebase in GitHub, it's useful for description, usage, parameters, etc; the readme's are also copied to the [documentation](https://tanzucommunityedition.io/docs/v0.12/) site for reference. If we don't have a readme, I think that we will end up with a dead link in the left nav menu for packages on that page.

seemiller commented:
Although, I just noticed that we are not exposing secretgen-controller in the user-managed package repo, so those docs will not show up on the documentation site anyway.

neil-hickey commented:
@seemiller Makes sense! Thanks for the background, so is it fine to have a README at the top level that is not version specific?

neil-hickey committed:
Add readme

Signed-off-by: Neil Hickey <nhickey@vmware.com>
